src/bp/core/repositories/conversations.ts

```diff
@@ -0,0 +1,238 @@
+import * as sdk from 'botpress/sdk'
+import Database from 'core/database'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+import uuid from 'uuid'
+
+const conversationsTable = 'web_conversations'
+const messageTable = 'web_messages'
+
+@injectable()
+export class ConversationsRepository {
+  constructor(@inject(TYPES.Database) private database: Database) {}
+
+  async getOrCreateConversation(
+    botId: string,
+    userId: string,
+    opt: { lifetime?: string; isUserCreated?: boolean } = { lifetime: '6 hours', isUserCreated: false }
+  ): Promise<any> {
+    const lessThanLifetime = this.database.knex.date.isAfter(
+      'last_heard_on',
+      moment()
+        .subtract(ms(opt.lifetime), 'ms')
+        .toDate()
+    )
+
+    const conversation = await this.database
+      .knex(conversationsTable)
+      .select('id')
+      .whereNotNull('last_heard_on')
+      .andWhere({ userId, botId })
+      .andWhere(lessThanLifetime)
+      .orderBy('last_heard_on', 'desc')
+      .limit(1)
+      .get(0)
+
+    return conversation ? conversation['id'] : this.createConversation(botId, userId, opt.isUserCreated)
```

Comment:
hmm in the case where createConversation is returned, only the id will be returned and it seems to me like if you want to get the conversation, you want the full object and not just the id ...